### PR TITLE
feat(provider/kubernetes): deploy daemonset and other fixes

### PR DESCRIPTION
**Pull request by chlung**
About this PR,
- support to deploy daemonset
- fix vpc creation problem and others
- refactor updateStrategy for daemonset and statefulset

Thanks.

We prefer small, well tested pull requests.

Please refer to [Contributing to Spinnaker](https://spinnaker.io/community/contributing/).

When filling out a pull request, please consider the following:

* Follow the commit message conventions [found here](http://www.spinnaker.io/v1.0/docs/how-to-submit-a-patch).
* Provide a descriptive summary for your changes.
* If it fixes a bug or resolves a feature request, be sure to link to that issue.
* Add inline code comments to changes that might not be obvious.
* Squash your commits as you keep adding changes.
* Add a comment to @spinnaker/reviewers for review if your issue has been outstanding for more than 3 days.

Note that we are unlikely to accept pull requests that add features without prior discussion. The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

### CODE REVIEW

**Comment by chlung** [comment]
@lwander Please review.  There are a lot of changes for code cleanup and refactor.  Thanks.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiConverter.groovy [hunk 1]
```diff
@@ -349,8 +356,8 @@ class KubernetesClientApiConverter {
     def persistentVolumeClaims = new ArrayList<V1PersistentVolumeClaim>()
     if (description.volumeClaims) {
       description.volumeClaims.forEach({ claim ->
-        V1PersistentVolumeClaimSpec spec = new V1PersistentVolumeClaimSpec()
-        V1ObjectMeta metadata = new V1ObjectMeta()
+        def spec = new V1PersistentVolumeClaimSpec()
```
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiConverter.groovy]: Why replace the type with `def`?

Comment (chlung) [reply clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesClientApiConverter.groovy]: No particular reasons.  I just think that the code already initialized with the type.

**Comment by chlung** [comment]
@lwander thanks

**Pull request merged**